File: Cpp/650_2_Keys_Keyboard.cpp
class Solution {
public:
    int minSteps(int n) {
        vector<int> dp(n + 1);
        for(int i = 2; i <= n; ++i) {
            dp[i] = i;
        }
        
        for(int i = 2; i <= n; ++i) {
            for (int j = 2; i * j <= n; ++j) {
                dp[i * j] = min(dp[i * j], dp[i] + j); 
            }
        }
        
        return dp[n];
    }
};

File: Kotlin/650_2_Keys_Keyboard.kt
class Solution {
    fun minSteps(n: Int): Int {
        val dp = IntArray(n + 1) { it }
        dp[1] = 0
        
        for(i in 2..n) {
            var j = 2
            while(i * j <= n) {
                dp[i * j] = minOf(dp[i * j], dp[i] + j)
                ++j
            }
        }

        return dp[n]
    }
}